includes/REST_API/Stories_Controller.php

```diff
@@ -86,6 +86,28 @@ public function prepare_item_for_response( $post, $request ) {
 			$data['style_presets'] = is_array( $style_presets ) ? $style_presets : self::EMPTY_STYLE_PRESETS;
 		}
 
+		if ( in_array( 'preview_link', $fields, true ) ) {
+			// Based on https://github.com/WordPress/wordpress-develop/blob/8153c8ba020c4aec0b9d94243cd39c689a0730f7/src/wp-admin/includes/post.php#L1445-L1457
+			if ( 'draft' === $post->post_status || empty( $post->post_name ) ) {
+				$view_link = get_preview_post_link( $post );
+			} else {
+				if ( 'publish' === $post->post_status ) {
+					$view_link = get_permalink( $post );
+				} else {
+					if ( ! function_exists( 'get_sample_permalink' ) ) {
+						require_once ABSPATH . 'wp-admin/includes/post.php';
```

Comment:
Is this needed? Shouldn't this already be loaded?

Reply:
The REST API can be called from outside the admin context, so I think it is an issue. This is copied from core. 
https://github.com/WordPress/wordpress-develop/blob/9983be3248ea497ecb001be82c618c7849d3c51f/src/wp-includes/rest-api/endpoints/class-wp-rest-posts-controller.php#L1825-L1839

Reply:
Ah, realizing now that file resides in `wp-admin`, not `wp-includes`!